### PR TITLE
Support bytes type

### LINKED ISSUE #32: Support bytes
As `bytes("a string")` also requires an encoding, we would need an encoding metadata field

### PR DESCRIPTION
**Pull request by rogiervandergeer**
Closes #32

### CODE REVIEW

File: src/pydargs/__init__.py [hunk 1]
```diff
@@ -87,13 +87,13 @@ def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
...
-                union_parser = partial(_parse_union, union_type=field.type)
-                setattr(union_parser, "__name__", repr(field.type))
+                bytes_parser = partial(_parse_union, union_type=field.type)
+                setattr(bytes_parser, "__name__", repr(field.type))
                 parser.add_argument(
                     argument_name,
                     default=argparse.SUPPRESS,
                     help=f"Override field {field.name}.",
-                    type=union_parser,
+                    type=bytes_parser,
```
Comment (antonsteenvoorden) [inline src/pydargs/__init__.py]: are these renames intentional?

File: src/pydargs/__init__.py [hunk 2]
```diff
@@ -135,7 +135,17 @@ def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
                 type=lambda x: field.type[x],
                 **kwargs,
             )
-
+        elif field.type is bytes:
+            encoding = field.metadata.get("encoding", "utf-8")
+            bytes_parser = partial(field.type, encoding=encoding)
+            setattr(bytes_parser, "__name__", encoding)
```
Comment (antonsteenvoorden) [inline src/pydargs/__init__.py:141]: is this just for the error message again?

File: tests/test_types.py [hunk 3]
```diff
@@ -51,6 +51,30 @@ def test_second_positional(self):
...
+        z: int = 12
+
+    def test_default(self):
+        config = parse(self.Config, [])
+        assert config.a == b"a"
+        assert config.b == b"b"
+        assert config.z == 12
+
+    def test_encoding(self):
+        config = parse(self.Config, ["--a", "ꀀ"])
```
Comment (antonsteenvoorden) [inline tests/test_types.py:68]: 😂 I wiped my screen to get rid of the dust...

.. but it was part of the character

**Review by antonsteenvoorden — COMMENTED** [review]
nice, left 2 questions

File: src/pydargs/__init__.py (hunk 1 above)
Comment (rogiervandergeer) [reply src/pydargs/__init__.py]: They were not, thanks. Fixed!

File: src/pydargs/__init__.py (hunk 2 above)
Comment (rogiervandergeer) [reply src/pydargs/__init__.py:141]: Indeed, it places the 'ascii' in

```
assert "argument --b: invalid ascii value: 'ꀀ'\n" in captured.err
```

**Review by antonsteenvoorden — APPROVED** [review]
👍 nice

**Pull request merged**